libindy_vdr/src/resolver/resolver.rs

```diff
@@ -0,0 +1,203 @@
+use super::did::DidUrl;
+use crate::common::error::prelude::*;
+
+use crate::ledger::RequestBuilder;
+use crate::pool::{Pool, PoolRunner, RequestResult, TimingResult};
+
+use super::types::*;
+use super::utils::*;
+
+/// DID (URL) Resolver interface for a pool compliant with did:indy method spec
+/// The resolver interface is bound to a specific indy network and does not evaluate
+/// the namespace part of the DID. You need to create a resolver instance for each
+/// indy network you want to support.
+/// The `PoolResolver` uses async/await.
+pub struct PoolResolver<T: Pool> {
+    pool: T,
+}
+
+impl<T: Pool> PoolResolver<T> {
+    pub fn new(pool: T) -> PoolResolver<T> {
+        PoolResolver { pool }
+    }
+
+    /// Dereference a DID Url and return a serialized `DereferencingResult`
+    pub async fn dereference(&self, did_url: &str) -> VdrResult<String> {
+        debug!("PoolResolver: Dereference DID Url {}", did_url);
+        let did_url = DidUrl::parse(did_url)?;
+        let (data, metadata) = self._resolve(&did_url).await?;
+
+        let content = match data {
+            Result::Content(c) => Some(c),
+            _ => None,
+        };
+
+        let md = if let Metadata::ContentMetadata(md) = metadata {
+            Some(md)
+        } else {
+            None
+        };
+
+        let result = DereferencingResult {
+            dereferencing_metadata: None,
+            content_stream: content,
+            content_metadata: md,
+        };
+
+        Ok(serde_json::to_string_pretty(&result).unwrap())
+    }
+
+    /// Resolve a DID and return a serialized `ResolutionResult`
+    pub async fn resolve(&self, did: &str) -> VdrResult<String> {
+        debug!("PoolResolver: Resolve DID {}", did);
+        let did = DidUrl::parse(did)?;
+        let (data, metadata) = self._resolve(&did).await?;
+
+        let md = if let Metadata::DidDocumentMetadata(md) = metadata {
+            Some(md)
+        } else {
+            None
+        };
+
+        let diddoc = match data {
+            Result::DidDocument(mut doc) => {
+                // Try to find legacy endpoint using a GET_ATTRIB txn if diddoc_content is none
+                if doc.diddoc_content.is_none() {
+                    let (seq_no, timestamp) = if let Some(md) = &md {
+                        let j_seq_no = &md.node_response["result"]["seqNo"];
```

Comment:
More out of curiousity, what is the `j_` prefix?

Reply:
I think it's short for json, since they are json nodes.